src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -193,41 +193,80 @@ static SymbolNamesOption.NameParts GetParts(string name)
                     {
                         if (trimmedSuffix[i] != ' ')
                         {
-                            return new SymbolNamesOption.NameParts(split[0], trimmedSuffix);
+                            return new SymbolNamesOption<string>.NameParts(split[0], trimmedSuffix);
                         }
                     }
 
                     // Replace the special empty suffix symbol by an empty string
-                    return new SymbolNamesOption.NameParts(split[0], string.Empty);
+                    return new SymbolNamesOption<string>.NameParts(split[0], string.Empty);
                 }
 
-                return new SymbolNamesOption.NameParts(split[0], trimmedSuffix);
+                return new SymbolNamesOption<string>.NameParts(split[0], trimmedSuffix);
             }
         }
 
-        private static SymbolNamesOption GetSymbolNamesOption(
+        public static SymbolNamesOption<INamedTypeSymbol> GetAdditionalRequiredGenericInterfaces(
+            this AnalyzerOptions options,
+            DiagnosticDescriptor rule,
+            Compilation compilation,
+            CancellationToken cancellationToken)
+        {
+            return options.GetSymbolNamesOption(EditorConfigOptionNames.AdditionalRequiredGenericInterfaces, namePrefixOpt: "T:", rule, compilation, cancellationToken, x => GetParts(x, compilation));
+
+            static SymbolNamesOption<INamedTypeSymbol>.NameParts GetParts(string name, Compilation compilation)
+            {
+                var split = name.Split(new[] { "->" }, StringSplitOptions.RemoveEmptyEntries);
+
+                // If we don't find exactly one '->', we assume that there is no given suffix.
+                if (split.Length != 2)
+                {
+                    return new SymbolNamesOption<INamedTypeSymbol>.NameParts(name);
+                }
+
+                var genericInterfaceFullName = split[1].Trim();
+                if (!genericInterfaceFullName.StartsWith("T:", StringComparison.Ordinal))
+                {
+                    genericInterfaceFullName = $"T:{genericInterfaceFullName}";
+                }
+
+                var matchingSymbols = DocumentationCommentId.GetSymbolsForDeclarationId(genericInterfaceFullName, compilation);
+
+                if (matchingSymbols.Length != 1 ||
+                    !(matchingSymbols[0] is INamedTypeSymbol namedType) ||
+                    namedType.TypeKind != TypeKind.Interface ||
+                    !namedType.IsGenericType)
+                {
+                    // Invalid matching type so we assume there was no associated type
```

Comment:
We may want to eventually report a diagnostic here so user can fix his configuration. It is fine to not do it here as it seems applicable for all our option values.

Reply:
#3267